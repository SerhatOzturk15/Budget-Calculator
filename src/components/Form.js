import React from "react";
import { MdSend } from "react-icons/md";
const Form = ({
  charge,
  amount,
  handleCharge,
  handleAmount,
  handleSubmit,
  edit
}) => {
  return (
    <form onSubmit={handleSubmit}>
      <div className="form-center">
        <div className="form-group">
          <label htmlFor="expense">charge</label>
          <input
            data-testid="form-charge-input"
            type="text"
            className="form-control"
            id="charge"
            name="charge"
            placeholder="e.g. credits"
            value={charge}
            onChange={handleCharge}
          />
        </div>
        <div className="form-group">
          <label htmlFor="expense">amount</label>
          <input
            data-testid="form-amount-input"
            type="number"
            className="form-control"
            id="amount"
            name="amount"
            placeholder="e.g. 10"
            value={amount}
            onChange={handleAmount}
          />
        </div>
      </div>
      <button type="submit" className="btn" data-testid="form-submit-button">
        {edit ? "edit" : "submit"}
        <MdSend className="btn-icon"></MdSend>
      </button>
    </form>
  );
};

export default Form;
