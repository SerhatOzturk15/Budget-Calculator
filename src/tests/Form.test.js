import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import Form from '../components/Form';


const setup = (props = {}) => {
    const component = shallow(<Form {...props}/>);
    return component;
}

describe('Alert component', () => {
    let component = setup();

    it('component should render', () => {
        const wrapper = component.find('.form-center');
        expect(wrapper.length).toBe(1);
    });

    it('snapshot should match', () => {
        const wrapper = component.find('.form-center');
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('component should render charge input', () => {
        const wrapper = component.find('[data-testid="form-charge-input"]');
        expect(wrapper.length).toBe(1);
    });

    it('component should render amount input', () => {
        const wrapper = component.find('[data-testid="form-amount-input"]');
        expect(wrapper.length).toBe(1);
    });
    

})