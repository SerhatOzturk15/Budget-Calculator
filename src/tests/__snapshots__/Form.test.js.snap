// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component snapshot should match 1`] = `
<div
  className="form-center"
>
  <div
    className="form-group"
  >
    <label
      htmlFor="expense"
    >
      charge
    </label>
    <input
      className="form-control"
      data-testid="form-charge-input"
      id="charge"
      name="charge"
      placeholder="e.g. credits"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="expense"
    >
      amount
    </label>
    <input
      className="form-control"
      data-testid="form-amount-input"
      id="amount"
      name="amount"
      placeholder="e.g. 10"
      type="number"
    />
  </div>
</div>
`;
